libraries/botbuilder-lg/src/activityFactory.ts

```diff
@@ -0,0 +1,361 @@
+/**
+ * @module botbuilder-lg
+ */
+/**
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import { Activity, SuggestedActions, Attachment, ActivityTypes, ActionTypes, CardAction } from 'botframework-schema';
+import { MessageFactory, CardFactory } from 'botbuilder-core';
+
+export class ActivityFactory {
+    private static GenericCardTypeMapping: Map<string, string> = new Map<string, string>
+    ([
+        [ 'herocard', CardFactory.contentTypes.heroCard ],
+        [ 'thumbnailcard', CardFactory.contentTypes.thumbnailCard ],
+        [ 'audiocard', CardFactory.contentTypes.audioCard ],
+        [ 'videocard', CardFactory.contentTypes.videoCard ],
+        [ 'animationcard', CardFactory.contentTypes.animationCard ],
+        [ 'signincard', CardFactory.contentTypes.signinCard ],
+        [ 'oauthcard', CardFactory.contentTypes.oauthCard ],
+    ]);
+
+    public static CreateActivity(lgStringResult: string): Activity {
+        let lgStructuredResult: any;
+        try {
+            lgStructuredResult = JSON.parse(lgStringResult);
+        } catch (error) {
+            return this.buildActivityFromText(lgStructuredResult.trim());
+        }
+
+        return this.buildActivityFromLGStructuredResult(lgStructuredResult);
+    }
+
+    private static buildActivityFromText(text: string): Activity {
+        return MessageFactory.text(text, text) as Activity;
+    }
+
+    private static buildActivityFromLGStructuredResult(lgJObj: any): Activity {
+        let activity: any = {};
+
+        const type: string = this.getStructureType(lgJObj);
+        if (ActivityFactory.GenericCardTypeMapping.has(type)) {
+            const attachment: Attachment = this.getAttachment(lgJObj);
+            if (attachment !== undefined) {
+                activity = MessageFactory.attachment(attachment);
+            }
```

Comment:
>} [](start = 12, length = 1)

Do you really expect to have an empty activity returned if you can't build an attachment? Seems like this should be an error.